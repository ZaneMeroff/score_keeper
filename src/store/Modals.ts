// Decorators
import { Action, Module, Mutation, VuexModule } from 'vuex-class-modules'

@Module
export default class Modals extends VuexModule {

  addPlayersModal: boolean = false
  clearDataModal: boolean = false
  deletePlayersModal: boolean = false
  resetScoreModal: boolean = false
  scoreLimitsModal: boolean = false

  get getShowAddPlayersModal(): boolean {
    return this.addPlayersModal
  }

  get getShowClearDataModal(): boolean {
    return this.clearDataModal
  }

  get getShowDeletePlayersModal(): boolean {
    return this.deletePlayersModal
  }

  get getShowResetScoreModal(): boolean {
    return this.resetScoreModal
  }

  get getShowScoreLimitsModal(): boolean {
    return this.scoreLimitsModal
  }

  // ------------------------------------------
  //       add players modal visibility
  // ------------------------------------------
  
  @Mutation
  addPlayersModalVisibility(show: boolean) {
    this.addPlayersModal = show
  }
  
  @Action
  action_addPlayersModalVisibility(show: boolean) {
    this.addPlayersModalVisibility(show)
  }

  // ------------------------------------------
  //        clear data modal visibility
  // ------------------------------------------
  
  @Mutation
  clearDataModalVisibility(show: boolean) {
    this.clearDataModal = show
  }
  
  @Action
  action_clearDataModalVisibility(show: boolean) {
    this.clearDataModalVisibility(show)
  }

  // ------------------------------------------
  //  confirm delete players modal visibility
  // ------------------------------------------
  
  @Mutation
  deletePlayersModalVisibility(show: boolean) {
    this.deletePlayersModal = show
  }

  @Action
  action_deletePlayersModalVisibility(show: boolean) {
    this.deletePlayersModalVisibility(show)
  }

  // ------------------------------------------
  //    confirm reset score modal visibility
  // ------------------------------------------
  
  @Mutation
  resetScoreModalVisibility(show: boolean) {
    this.resetScoreModal = show
  }

  @Action
  action_resetScoreModalVisibility(show: boolean) {
    this.resetScoreModalVisibility(show)
  }

  // ------------------------------------------
  //        score limit modal visibility
  // ------------------------------------------
  
  @Mutation
  scoreLimitModalVisibility(show: boolean) {
    this.scoreLimitsModal = show
  }

  @Action
  action_scoreLimitModalVisibility(show: boolean) {
    this.scoreLimitModalVisibility(show)
  }
}