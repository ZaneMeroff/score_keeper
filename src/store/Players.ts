// Vue & Decorators
import Vue from 'vue'
import { Action, Module, Mutation, VuexModule } from 'vuex-class-modules'

// Types
import { Player, PlayerData } from '@/types/players'

@Module
export default class Players extends VuexModule {

    playerData: PlayerData = {}

    get getPlayerData() {
      return this.playerData
    }

    // ------------------------------------------
    //              create player
    // ------------------------------------------

    @Mutation
    createPlayer(payload: Player) {
      Vue.set(this.playerData, payload.id, payload)
    }

    @Action
    action_createPlayer(payload: Player) {
      this.createPlayer(payload)
    }

    // ------------------------------------------
    //         modify target player score
    // ------------------------------------------

    @Mutation
    setPlayerScore(payload: { id: string, score: number }) {
      this.playerData[payload.id].score = payload.score
    }

    @Action
    action_setPlayerScore(payload: { id: string, score: number }) {
      this.setPlayerScore(payload)
    }

    // ------------------------------------------
    //         modify target player name
    // ------------------------------------------

    @Mutation
    setPlayerName(payload: { id: string, name: string }) {
      this.playerData[payload.id].name = payload.name
    }

    @Action
    action_setPlayerName(payload: { id: string, name: string }) {
      this.setPlayerName(payload)
    }

    // ------------------------------------------
    //        zero scores for all players
    // ------------------------------------------

    @Mutation
    zeroScores() {
      for (const player in this.playerData) {
        this.playerData[player].score = 0
      }
    }

    @Action
    action_zeroScores() {
      this.zeroScores()
    }
}