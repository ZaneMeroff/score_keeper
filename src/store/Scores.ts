// Decorators
import { Action, Module, Mutation, VuexModule } from 'vuex-class-modules'

// Types
import { Score, ScoreData } from '@/types/scores'
import Vue from 'vue'

@Module
export default class Scores extends VuexModule {

    scoreData: ScoreData = {}

    get getScoreData() {
      return this.scoreData
    }

    @Mutation
    setScoreData(payload: { id: string, updatedScore: number }) {
      // create or update score data based on id
      Vue.set(this.scoreData, payload.id, payload.updatedScore)
    }

    
}