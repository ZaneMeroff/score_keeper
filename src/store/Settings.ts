// Decorators
import Vue from 'vue'
import { Action, Module, Mutation, VuexModule } from 'vuex-class-modules'

@Module
export default class Settings extends VuexModule {

  darkMode: boolean = true
  minScore: number = 0
  maxScore: number = 10
  showScoreLimitsModal: boolean = false
  // showMenuDrawer: boolean = false

  get getDarkMode() {
    return this.darkMode
  }

  get getMinScore() {
    return this.minScore
  }

  get getMaxScore() {
    return this.maxScore
  }

  get getShowScoreLimitsModal() {
    return this.showScoreLimitsModal
  }

  // get getShowMenuDrawer() {
  //   return this.showMenuDrawer
  // }
  
  // ------------------------------------------
  //             toggle dark mode
  // ------------------------------------------

  @Mutation
  setDarkMode() {
    this.darkMode = !this.darkMode
  }

  @Action
  action_setDarkMode() {
    this.setDarkMode()
  }

  // ------------------------------------------
  //            set score limits
  // ------------------------------------------

  @Mutation
  setScoreLimits(payload: { min: number, max: number }) {
    this.minScore = payload.min
    this.maxScore = payload.max
  }

  @Action
  action_setScoreLimits(payload: { min: number, max: number }) {
    this.setScoreLimits(payload)
  }

  // ------------------------------------------
  //       score limit modal visibility
  // ------------------------------------------
  
  @Mutation
  setScoreLimitModalVisibility(show: boolean) {
    this.showScoreLimitsModal = show
  }

  @Action
  action_setScoreLimitModalVisibility(show: boolean) {
    this.setScoreLimitModalVisibility(show)
  }

  // ------------------------------------------
  //         menu drawer visibility
  // ------------------------------------------
  
  // @Mutation
  // setMenuDrawerlVisibility(show: boolean) {
  //   this.showMenuDrawer = show
  // }

  // @Action
  // action_setMenuDrawerlVisibility(show: boolean) {
  //   this.setMenuDrawerlVisibility(show)
  // }
}