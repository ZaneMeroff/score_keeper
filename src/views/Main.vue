<template>

  <section>
    <player-card 
      v-for="(player, index) in statePlayers.getPlayerData" 
      :key="index"
      :playerId="player.id" 
    />
  </section>

</template>

<script lang="ts">

// Decorators
import { Component, Vue } from 'vue-property-decorator'

// Vuex
import { statePlayers } from '@/store/index'

// Components
import PlayerCard from '@/components/PlayerCard.vue'

@Component({
  components: {
    'player-card' : PlayerCard,
  }
})
export default class Main extends Vue {

  statePlayers = statePlayers
}
</script>

<style>

  section {
    background-color: #151515;
    display: flex !important;
    flex-direction: column;
    height: 100vh;
    padding: 60px 20px 10px 20px;
    width: 100%;
  }

</style>
