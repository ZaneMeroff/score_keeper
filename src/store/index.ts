import Vue from 'vue'
import Vuex from 'vuex'
import VuexPersistence from 'vuex-persist'
import localforage from 'localforage'

import Modals from './Modals'
import Players from './Players'
import Settings from './Settings'

Vue.use(Vuex)

const vuexLocal = new VuexPersistence({
  storage: localforage as any,
  asyncStorage: true,
  modules: [ 'Modals', 'Players', 'Settings' ],
});

export const store: any = new Vuex.Store({
  // strict: process.env.DEV === 'true',
  state: {},
  modules: {},
  plugins: [ vuexLocal.plugin ],
})

export const stateModals= new Modals({ store, name: 'Modals' })
export const statePlayers= new Players({ store, name: 'Players' })
export const stateSettings = new Settings({ store, name: 'Settings' })
