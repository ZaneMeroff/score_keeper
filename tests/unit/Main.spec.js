import { createLocalVue, shallowMount } from '@vue/test-utils'
import { store } from "@/store/index"
import Main from '@/views/Main.vue'
import Vuex from "vuex"

// --------- Vuex setup ------------

const localVue = createLocalVue()
localVue.use(Vuex)

// ------------------------------------------

describe('Main', () => {

  describe("snapshots", () => {

    it("should render with default html", () => {
      const component = shallowMount(Main)

      expect(component.element).toMatchSnapshot()
    })

    // it("should render with a player-card", () => {
    //   const component = shallowMount(Main)

    //   // store.state.Players.action_createPlayers(1)
    //   store.state.Players.playerData = {
    //     "5abc23b2-c180-4bca-8287-a4785b7c5a9c": {
    //       id: "5abc23b2-c180-4bca-8287-a4785b7c5a9c",
    //       name: "Player 1",
    //       score: 0,
    //       color: "#ff3021"
    //     }
    //   }

    //   expect(component.element).toMatchSnapshot()
    // })
  })

  // it('test it', () => {
  //   const msg = 'new message'
  //   const wrapper = shallowMount(Main, {
  //     propsData: { msg }
  //   })
  //   expect(wrapper.text()).toMatch(msg)
  // })
})
