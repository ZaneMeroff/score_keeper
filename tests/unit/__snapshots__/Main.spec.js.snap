// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main snapshots should render with default html 1`] = `
<section
  style="background-color: rgb(62, 62, 62);"
>
  <div
    class="error-wrapper"
  >
    <p
      class="error-text"
      style="color: rgb(255, 255, 255);"
    >
      Players List is Empty
    </p>
     
    <p
      class="error-text"
      style="color: rgb(255, 255, 255);"
    >
      Select 'Add Players' from the App Menu
    </p>
  </div>
</section>
`;
